airbyte-cdk/bulk/toolkits/load-object-storage/src/main/kotlin/io/airbyte/cdk/load/command/object_storage/ObjectStoragePathConfiguration.kt

```diff
@@ -8,6 +8,7 @@ data class ObjectStoragePathConfiguration(
     val prefix: String,
     val pathPattern: String?,
     val fileNamePattern: String?,
+    val resolveNamesMethod: ((String) -> String)? = null
```

Comment:
I think this works? then we can avoid the nullable transform (which is kind of misleading, since you might expect `null` to be a noop transform rather than the actual `s3chars` thing) + have the ObjectStoragePathFactory call the method directly instead of wrapping in `transformName`
```suggestion
    val resolveNamesMethod: ((String) -> String) = { Transformations.toS3SafeCharacters(it) }
```